atlasdb-processors-tests/src/test/java/com/palantir/processors/AutoDelegateGenericTests.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * (c) Copyright 2018 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.processors;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.hasItems;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.lang.reflect.Modifier;
+import java.util.Set;
+
+import org.junit.Test;
+
+public class AutoDelegateGenericTests {
+    @Test
+    public void generatedInterfaceHasInterfaceMethods() {
+        Set<String> generatedMethods = TestingUtils.extractMethods(AutoDelegate_GenericsTester.class);
+        Set<String> originalMethods = extractNonStaticMethods(GenericsTester.class);
+
+        assertThat(generatedMethods, hasItems(originalMethods.toArray(new String[0])));
```

Comment:
please use assertJ for this:

```java
assertThat(generatedMethods.contains(originalMethods);
```

Reply:
Ah, ok. I copied this from the other classes (where I'd prefer to stick with old-style as I'm not feeling up to changing all the old tests to use AssertJ), but no reason to hang on to this for new classes.

Reply:
nit: we want `containsAll`, `contains` is varargs on elements